src/backend/distributed/operations/shard_split.c

```diff
@@ -1698,6 +1707,33 @@ ExecuteSplitShardReplicationSetupUDF(WorkerNode *sourceWorkerNode,
 }
 
 
+/*
+ * ExecuteSplitShardReleaseSharedMemory releases dynamic shared memory
+ * at source node.
+ * As a part of non-blocking split workflow, worker_split_shard_replication_setup allocates
+ * shared memory to store split information. This has to be released after split completes(or fails).
+ */
+static void
+ExecuteSplitShardReleaseSharedMemory(WorkerNode *sourceWorkerNode)
+{
+	char *superUser = CitusExtensionOwnerName();
```

Comment:
In the future we should maybe pass a workflow ID to worker_split_shard_release_dsm, use the current user, and check whether the current user matches the workflow ID owner.

Reply:
Yes, sure.